### PR TITLE
Add Recommendation shelf to Product Page

### PR DESCRIPTION
**Pull request by lariciamota**

### CODE REVIEW

File: packages/gatsby-theme-vtex/src/sdk/shelf/useRecommendationShelf.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+export const query = gql`
+  query ProductRecommendationsQuery($value: ID!) {
+    vtex {
+      productRecommendations(
+        identifier: { field: id, value: $value }
+        type: viewAndBought
+      ) {
+        productId
```
Comment (lucis) [inline packages/gatsby-theme-vtex/src/sdk/shelf/useRecommendationShelf.ts:36]: Do we have a Fragment for that so you can reuse?

Comment (tlgimenes) [reply packages/gatsby-theme-vtex/src/sdk/shelf/useRecommendationShelf.ts:36]: it's the productSummary fragment, right ?

File: packages/gatsby-theme-vtex/src/components/Shelf/index.tsx [hunk 2]
```diff
@@ -45,9 +47,11 @@ const Shelf: FC<Props> = ({
     autoplay,
   })
 
+  showArrows = showArrows && products.length >= Math.max(...pageSizes)
+
   return (
     <Fragment>
-      {title ? <ShelfTitle title={title} /> : null}
+      {title ? <ShelfTitle title={title} variant={titleVariant} /> : null}
```
Comment (tlgimenes) [inline packages/gatsby-theme-vtex/src/components/Shelf/index.tsx:54]: why is it called titleVariant and not only variant ? I think the best way of using variants in here is creating a variable called `variant` and passing `${variant}.title` as the `<ShelfTitle />` component variant

File: packages/gatsby-theme-vtex/src/components/Shelf/index.tsx [hunk 3]
```diff
@@ -45,9 +47,11 @@ const Shelf: FC<Props> = ({
     autoplay,
   })
 
+  showArrows = showArrows && products.length >= Math.max(...pageSizes)
+
   return (
     <Fragment>
-      {title ? <ShelfTitle title={title} /> : null}
+      {title ? <ShelfTitle title={title} variant={`${variant}.title`} /> : null}
```
[the remark this answers is not in the document]
Comment (lariciamota) [reply packages/gatsby-theme-vtex/src/components/Shelf/index.tsx:54]: Awesome, I'll do it!

**Pull request merged**